lib/library.cpp

```diff
@@ -107,8 +107,9 @@ Library::Error Library::load(const char exename[], const char path[])
         cfgfolders.emplace_back(FILESDIR "/cfg");
 #endif
         if (exename) {
-            const std::string exepath(Path::fromNativeSeparators(Path::getPathFromFilename(exename)));
+            const std::string exepath(Path::fromNativeSeparators(Path::getPathFromFilename(*exename ? exename : Path::getCurrentExecutablePath())));
```

Comment:
I think it would be better to pass `exename` to `getCurrentExecutablePath()` and then it can use that path as a fallback for unknown OS or when an error occures.